drivers/scheduler/k8s/k8s.go

```diff
@@ -345,13 +340,16 @@ func (k *K8s) ParseSpecs(specDir, storageProvisioner string) ([]interface{}, err
 	return specs, nil
 }
 
-func isValidProvider(specPath string) bool {
-	for _, driver := range volume.GetVolumeDrivers() {
-		if strings.Contains(specPath, "/"+driver+"/") { // Check for directories for cloud providers, ignore files
-			return true
+func isValidProvider(specPath, storageProvisioner string) bool {
+	// Skip all storage provisioner specific spec except storageProvisioner
+	for _, driver := range volume.GetVolumeProvisioners() {
+		if driver != storageProvisioner &&
```

Comment:
this would return false if driver = pxd and storageProvisioner = portworx, when it should return true

Reply:
I guess driver variable should be renamed to provisioner. I am building spec loading based on provisioner type(default porworx). I am trying to figureout with @disrani-px why is it a bad idea.

Reply:
@Rohit-PX I've changed that, now specs are loaded based on volume driver.